G3D10/G3D-app.lib/source/TriTreeBase.cpp

```diff
@@ -140,9 +140,10 @@ void TriTreeBase::intersectRays
         for (size_t i = start; i < end; ++i) {
             const Hit& hit = pHit[i];
             if (hit.triIndex != Hit::NONE) {
-                // TODO: compute MIP level using coherence, pass to sample
-                pTri[hit.triIndex].sample(hit.u, hit.v, hit.triIndex, m_vertexArray, hit.backface, pSurfel[i]);
-            } else {
+                // Pass twoSided to determine whether or not to flip normals.
+                pTri[hit.triIndex].sample(hit.u, hit.v, hit.triIndex, m_vertexArray, hit.backface, pSurfel[i], pTri[hit.triIndex].twoSided());
```

Comment:
If I'm reading this right, the `pTri[hit.triIndex].twoSided()` is actually being mapped to the `du` argument instead of the intended `twoSided` argument. Here's the call signature for `Tri::sample()`

`void Tri::sample(float u, float v, int triIndex, const CPUVertexArray& vertexArray, bool backface, shared_ptr<Surfel>& surfel, float du, float dv, bool twoSided) const`